### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by Lotes**
* Added a helper function for asserting type emptiness (never)
* Added better error message when was needed

### CODE REVIEW

**Comment by Lotes** [comment]
See #462

File: packages/langium/src/utils/errors.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { CstNode } from '..';
```
Comment (msujew) [inline packages/langium/src/utils/errors.ts:7]: Minor: Please don't import from the index (could easily lead to cyclic dependency issues), but from the actual file.

File: packages/langium/src/utils/errors.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export class ErrorWithLocation extends Error {
+    constructor(node: CstNode|undefined, message: string) {
+        super(`${message} at ${node?.range.start.line || 0}:${node?.range.start.character || 0}`);
+    }
+}
+
+export function assertUnreachable(_: never): never {
+    throw new Error('Error! Please fulfill the logic also for the other types in the inheritence hierarchy.');
```
Comment (msujew) [inline packages/langium/src/utils/errors.ts:16]: Minor: The error message is a bit too specific for my taste. This error can happen anytime we get an unexpected/unhandled value in an `if-else/switch` block, so I imagine something like this:

```suggestion
    throw new Error('Error! The input value was not handled.');
```

Comment (Lotes) [reply packages/langium/src/utils/errors.ts:16]: `assertUnreachable` is a compile-time trick and will lead to an error, when the parameter is not the empty type. So actually we never reach this point.

File: packages/langium/src/utils/errors.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+}
```
[the remark this answers is not in the document]
Comment (msujew) [reply packages/langium/src/utils/errors.ts]: GitHub showed me that the file didn't end on a newline. But you seemed to have fixed it already 👍

File: packages/langium/src/utils/errors.ts (hunk 2 above)
Comment (msujew) [reply packages/langium/src/utils/errors.ts]: I see, alright then.

**Review by msujew — APPROVED** [review]
Thanks, LGTM 👍

**Pull request merged**